outlook/src/taskpane/components/Company/CompanySection/CompanySection.tsx

```diff
@@ -106,50 +107,64 @@ class CompanySection extends React.Component<CompanySectionProps, CompanySection
 
     }
 
-    private getCompanyLayout= () => {
-        const company = this.state.company as Company;
-
-        if (company.isEmpty())
+    private updateCompanyRequest = () => {
+        if (!this.props.partner.email)
         {
-            let enrichAndCreate = null;
-            if (this.props.partner.isAddedToDatabase() && !this.enriched)
+            let enrichmentInfo = new EnrichmentInfo(EnrichmentInfoType.EnrichContactWithNoEmail,
+                _t("This contact has no email address, no company could be enriched."));
+            this.context.showTopBarMessage(enrichmentInfo);
+            return;
+        }
+
+        this.setState({isLoading: true});
+
+
+        const enrichRequest = sendHttpRequest(HttpVerb.POST, api.baseURL + api.enrichAndUpdateCompany,
+            ContentType.Json, this.context.getConnectionToken(), {
+                partner_id: this.props.partner.id,
+            }, true);
+
+        this.context.addRequestCanceller(enrichRequest.cancel);
+
+        enrichRequest.promise.then((response) => {
+            const parsed = JSON.parse(response);
+
+            if (parsed.result['enrichment_info'])
             {
-                enrichAndCreate = (
-                    <div>
-                        <div>
-                            {_t("No company attached to this contact")}
-                        </div>
-                        <div className="odoo-secondary-button" style={{margin: "8px auto auto auto"}} onClick={this.enrichCompanyForPartnerRequest}>
-                            {_t("Create a Company")}
-                        </div>
-                    </div>
-                );
+                const enrichmentInfo = new EnrichmentInfo(parsed.result['enrichment_info'].type,
+                    parsed.result['enrichment_info'].info);
+                this.context.showTopBarMessage(enrichmentInfo);
+
+                if (enrichmentInfo.type == EnrichmentInfoType.NoData)
+                    this.enriched = true;
+            }
+
+            if (parsed.result['company'])
+            {
+                this.enriched = true;
+                const company = Company.fromJSON(parsed.result.company);
+                this.enriched = true;
```

Comment:
huh?

Reply:
Yes :) this does not make sense, it has now been corrected.